docs/the-new-architecture/pure-cxx-modules.md

```diff
@@ -0,0 +1,472 @@
+2. Inside the `shared` folder, create a new file called `NativeSampleModule.h`.
+
+   ```cpp
+   #pragma once
+
+   #include <AppSpecsJSI.h>
+
+   #include <memory>
+   #include <string>
+
+   namespace facebook::react {
+
+   class NativeSampleModule : public NativeSampleModuleCxxSpec<NativeSampleModule> {
+   public:
+     NativeSampleModule(std::shared_ptr<CallInvoker> jsInvoker);
+
+     std::string reverseString(jsi::Runtime& rt, std::string input);
+   };
+
+   } // namespace facebook::react
+
+   ```
+
+3. Inside the `shared` folder, create a new file called `NativeSampleModule.cpp`.
+
+   ```cpp
+   #include "NativeSampleModule.h"
+
+   namespace facebook::react {
+
+   NativeSampleModule::NativeSampleModule(std::shared_ptr<CallInvoker> jsInvoker)
+       : NativeSampleModuleCxxSpec(std::move(jsInvoker)) {}
+
+   std::string NativeSampleModule::reverseString(jsi::Runtime& rt, std::string input) {
+     return std::string(input.rbegin(), input.rend());
+   }
+
+   } // namespace facebook::react
+   ```
+
+Let's have a look at the two files we created.
+
+The `NativeSampleModule.h` file is the header file for a Pure C++ TurboModule. The `include` statements make sure that we include the specs that will be created by Codegen and that contains the interface and the base class we need to implement.
+
+The module lives in the `facebook::react` namespace to have access to all the types that live in that namespace.
+
+The class `NativeSampleModule` is the actual Turbo Native Mdoule class and it extends the `NativeSampleModuleCxxSpec` class which contains some glue code and boilerplate code to let this class behave as a Turbo Native Module.
+
+Finally, we have the constructor, that accepts a pointer to the `CallInvoker`, to communicate with JS if needed and the function's prototype we have to implement.
+
+The `NativeSampleModule.cpp` files is the actual implementation of our Turbo Native Module and implements the constructor and the method that we declared in the specs.
+
+## 4. Register the Module in the platform
+
+The next steps will let us register the module in the platform. This is the step that exposes the native code to JS so that the React Native application can finally call the native methods from the JS layer.
+
+This is the only time when we will have to write some platform-specific code.
+
+### Android
+
+To make sure that the iOS app can effectively build the C++ Turbo Native Module, we need to:
+
+1. Create a `CMakeLists.txt` to access our C++ code
+2. Modify `gradle.build` to point to the newly created `CMakeLists.txt` file.
+3. Create an `onLoad.cpp` file in our Android app to register the new Turbo Native Module.
+
+#### 1. Create the `CMakeList` file
+
+Android uses CMake to build. CMake needs to access the files we defined in our shared folder, to be able to build them.
+
+1. Create a new folder `SampleApp/android/app/src/main/jni`. The `jni` folder is where the C++ side of Android lives
```

Comment:
Does your path always start with `SampleApp/`? I would rather start with `./` here

Reply:
The guide starts with a command that creates the `SampleApp` using the CLI. I feel that repeating `SampleApp` gives the user an anchor to understand where to start looking. otherwise `./android` can be ambiguous and might raise questions on what is the starting point.
But I'm okay to change it, if we disagree.

Reply:
No strong opinion here as long as we're consistent

Reply:
yep, looks like we are consistent. I check all the usages of `/`